// --------------------------------------------------------------------------------
// SPDX-FileCopyrightText: 2024-2025 Martin Jan Köhler and Harald Pretl
// Johannes Kepler University, Institute for Integrated Circuits.
//
// This file is part of KPEX 
// (see https://github.com/iic-jku/klayout-pex).
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program. If not, see <http://www.gnu.org/licenses/>.
// SPDX-License-Identifier: GPL-3.0-or-later
// --------------------------------------------------------------------------------
syntax = "proto3";

package kpex.geometry;

// NOTE: all coordinates are in database units, (not floating point coordinates in µm)

message Point {
    int64 x = 1;
    int64 y = 2;

    string net = 10; // property
}

message Box {
    Point lower_left = 1;
    Point upper_right = 2;

    string net = 10; // property
}

message Polygon {
    repeated Point hull_points = 1;

    string net = 10; // property
}

message Region {
    repeated Polygon polygons = 1;
}
